ssl/statem/statem_clnt.c

```diff
@@ -1730,9 +1730,10 @@ MSG_PROCESS_RETURN tls_process_server_certificate(SSL *s, PACKET *pkt)
             }
             if (!tls_collect_extensions(s, &extensions,
                                         SSL_EXT_TLS1_3_CERTIFICATE, &rawexts,
-                                        &al, NULL)
+                                        &al, NULL, chainidx == 0)
                     || !tls_parse_all_extensions(s, SSL_EXT_TLS1_3_CERTIFICATE,
-                                                 rawexts, x, chainidx, &al)) {
+                                                 rawexts, x, chainidx, &al,
+                                                 !PACKET_remaining(pkt))) {
```

Comment:
PACKET_remainin()'s result is not a boolean, so this check should probably be " == 0"

It's also a little bit of spooky-action-at-a-distance that could break if someone adds code touching pkg between where &extensions is populated and here, but I think I can live with it.